pkg/api/rest/create.go

```diff
@@ -57,7 +57,14 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx api.Context, obj runtime.Obje
 			return errors.NewBadRequest("the namespace of the provided object does not match the namespace sent on the request")
 		}
 	} else {
-		objectMeta.Namespace = api.NamespaceNone
+		if ns, ok := api.NamespaceFrom(ctx); ok && ns != api.NamespaceNone {
+			return errors.NewBadRequest("namespace is not allowed on this type")
```

Comment:
> Please note that some types (events) can have both namespaced and non-namespaced objects.

Doesn't look like the event registry is setting a CreateStrategy or UpdateStrategy, so this isn't used for Events. If Events had been using this previously, though, it would have disallowed some events to be namespaced and some not...